### ISSUE TITLE
`update_or_create` doesn't work correctly in version 0.17.3

### ISSUE THREAD

**Issue by DavideGalilei** [tortoise/tortoise-orm#782]
## Describe the bug
`update_or_create` doesn't work as expected on a postgres database connection, if an object with `param` parameter (look at the first example) doesn't exist in the table, an `asyncpg.exceptions._base.InterfaceError` error will be raised. However, if it exists, the row will be correctly updated with no errors (look at the second example).
Although `get_or_create` works fine in creating objects whose don't exist, `update_or_create` doesn't.

## To Reproduce
To setup a quick postgres server I used docker:
```bash
$ docker run --rm --name pg-docker -e POSTGRES_PASSWORD=docker -d -p 5432:5432 -v $HOME/postgres-data:/var/lib/postgresql/data postgres
```
Then I installed tortoise-orm this way:
```bash
$ pip install -U tortoise-orm[asyncpg]==0.17.3
```

**First example:**
```python
import logging

from tortoise import Tortoise, fields, run_async
from tortoise.models import Model

logging.basicConfig(level=logging.DEBUG)

class Entry(Model):
    param = fields.TextField()
    name = fields.TextField()

async def main():
    await Tortoise.init(
        db_url="postgres://postgres:docker@localhost:5432/test",
        modules={"models": ["__main__"]},
        #_create_db=True  # Use it only once at first run
    )
    await Tortoise.generate_schemas()
    await Entry.all().delete()
    entry = await Entry.update_or_create(param="task1", defaults=dict(name="Task"))

run_async(main())
```
**Output:**
```python-traceback
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:tortoise:Tortoise-ORM startup
    connections: {'default': {'engine': 'tortoise.backends.asyncpg', 'credentials': {'port': 5432, 'database': 'test', 'host': 'localhost', 'user': 'postgres', 'password': 'do***'}}}
    apps: {'models': {'models': ['__main__'], 'default_connection': 'default'}}
DEBUG:db_client:Created connection pool <asyncpg.pool.Pool object at 0x7ff0ae3ed040> with params: {'host': 'localhost', 'port': 5432, 'user': 'postgres', 'database': 'test', 'min_size': 1, 'max_size': 5, 'connection_class': <class 'asyncpg.connection.Connection'>, 'loop': None}
DEBUG:tortoise:Creating schema: CREATE TABLE IF NOT EXISTS "entry" (
    "id" SERIAL NOT NULL PRIMARY KEY,
    "param" TEXT NOT NULL,
    "name" TEXT NOT NULL
);
DEBUG:db_client:CREATE TABLE IF NOT EXISTS "entry" (
    "id" SERIAL NOT NULL PRIMARY KEY,
    "param" TEXT NOT NULL,
    "name" TEXT NOT NULL
);
DEBUG:db_client:DELETE FROM "entry": None
DEBUG:db_client:SELECT "id","param","name" FROM "entry" WHERE "param"='task1' LIMIT 2 FOR UPDATE: None
DEBUG:db_client:SELECT "id","param","name" FROM "entry" WHERE "param"='task1' LIMIT 2: None
DEBUG:db_client:INSERT INTO "entry" ("param","name") VALUES ($1,$2) RETURNING "id": ['task1', 'Task']
DEBUG:db_client:Closed connection pool None with params: {'host': 'localhost', 'port': 5432, 'user': 'postgres', 'database': 'test', 'min_size': 1, 'max_size': 5, 'connection_class': <class 'asyncpg.connection.Connection'>, 'loop': None}
INFO:tortoise:Tortoise-ORM shutdown
Traceback (most recent call last):
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/models.py", line 1041, in get_or_create
    return await cls.filter(**kwargs).using_db(connection).get(), False
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/queryset.py", line 891, in _execute
    raise DoesNotExist("Object does not exist")
tortoise.exceptions.DoesNotExist: Object does not exist

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/davide/Desktop/bugreport.py", line 22, in <module>
    run_async(main())
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/__init__.py", line 706, in run_async
    loop.run_until_complete(coro)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/davide/Desktop/bugreport.py", line 20, in main
    entry = await Entry.update_or_create(param="task1", defaults=dict(name="Task"))
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/models.py", line 1082, in update_or_create
    return await cls.get_or_create(defaults, connection, **kwargs)
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/models.py", line 1044, in get_or_create
    return await cls.create(using_db=using_db, **defaults, **kwargs), True
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/models.py", line 1105, in create
    await instance.save(using_db=db, force_create=True)
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/models.py", line 926, in save
    await executor.execute_insert(self)
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/backends/base/executor.py", line 203, in execute_insert
    insert_result = await self.db.execute_insert(self.insert_query, values)
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/backends/asyncpg/client.py", line 36, in translate_exceptions_
    return await func(self, *args)
  File "/home/davide/.local/lib/python3.9/site-packages/tortoise/backends/asyncpg/client.py", line 141, in execute_insert
    return await connection.fetchrow(query, *values)
  File "/home/davide/.local/lib/python3.9/site-packages/asyncpg/pool.py", line 55, in call_con_method
    raise exceptions.InterfaceError(
asyncpg.exceptions._base.InterfaceError: cannot call Connection.fetchrow(): connection has been released back to the pool
```

**Second example:**
```python
import logging

from tortoise import Tortoise, fields, run_async
from tortoise.models import Model

logging.basicConfig(level=logging.DEBUG)

class Entry(Model):
    param = fields.TextField()
    name = fields.TextField()

async def main():
    await Tortoise.init(
        db_url="postgres://postgres:docker@localhost:5432/test",
        modules={"models": ["__main__"]},
        #_create_db=True  # Use it only once at first run
    )
    await Tortoise.generate_schemas()
    await Entry.all().delete()
    entry = await Entry.get_or_create(param="task1", defaults=dict(name="Task"))
    print((await Entry.get(param="task1")).name)
    await Entry.update_or_create(param="task1", defaults=dict(name="New Task Name"))
    print((await Entry.get(param="task1")).name)

run_async(main())
```
**Output:**
```python-traceback
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:tortoise:Tortoise-ORM startup
    connections: {'default': {'engine': 'tortoise.backends.asyncpg', 'credentials': {'port': 5432, 'database': 'test', 'host': 'localhost', 'user': 'postgres', 'password': 'do***'}}}
    apps: {'models': {'models': ['__main__'], 'default_connection': 'default'}}
DEBUG:db_client:Created connection pool <asyncpg.pool.Pool object at 0x7f7298ec1110> with params: {'host': 'localhost', 'port': 5432, 'user': 'postgres', 'database': 'test', 'min_size': 1, 'max_size': 5, 'connection_class': <class 'asyncpg.connection.Connection'>, 'loop': None}
DEBUG:tortoise:Creating schema: CREATE TABLE IF NOT EXISTS "entry" (
    "id" SERIAL NOT NULL PRIMARY KEY,
    "param" TEXT NOT NULL,
    "name" TEXT NOT NULL
);
DEBUG:db_client:CREATE TABLE IF NOT EXISTS "entry" (
    "id" SERIAL NOT NULL PRIMARY KEY,
    "param" TEXT NOT NULL,
    "name" TEXT NOT NULL
);
DEBUG:db_client:DELETE FROM "entry": None
DEBUG:db_client:SELECT "name","id","param" FROM "entry" WHERE "param"='task1' LIMIT 2: None
DEBUG:db_client:INSERT INTO "entry" ("param","name") VALUES ($1,$2) RETURNING "id": ['task1', 'Task']
DEBUG:db_client:SELECT "name","id","param" FROM "entry" WHERE "param"='task1' LIMIT 2: None
Task
DEBUG:db_client:SELECT "name","id","param" FROM "entry" WHERE "param"='task1' LIMIT 2 FOR UPDATE: None
DEBUG:db_client:UPDATE "entry" SET "param"=$1,"name"=$2 WHERE "id"=$3: ['task1', 'New Task Name', 5]
DEBUG:db_client:SELECT "name","id","param" FROM "entry" WHERE "param"='task1' LIMIT 2: None
New Task Name
DEBUG:db_client:Closed connection pool None with params: {'host': 'localhost', 'port': 5432, 'user': 'postgres', 'database': 'test', 'min_size': 1, 'max_size': 5, 'connection_class': <class 'asyncpg.connection.Connection'>, 'loop': None}
INFO:tortoise:Tortoise-ORM shutdown
```

## Expected behavior
`update_or_create` should just create the object in database (and should not throw any exception) if it doesn't exist, as it previously did in version [0.17.2](https://github.com/tortoise/tortoise-orm/releases/tag/0.17.2). Indeed, reverting to that version fixes the issue:
```bash
$ pip install -U tortoise-orm[asyncpg]==0.17.2
```

## Additional context
Issue doesn't seem to exist on an sqlite database connection.
**Postgres version (docker):** `psql (PostgreSQL) 13.3 (Debian 13.3-1.pgdg100+1)`
**Tortoise version:** [0.17.3](https://github.com/tortoise/tortoise-orm/releases/tag/0.17.3)
**asyncpg version:** `0.23.0`
**Python version:** `3.9.5`
**OS:** Arch Linux

**Comment by GodSaveTheDoge** [comment]
I was able to reproduce.
The "offending" commit is 04cc895630452d5b7101f6393b5bc54d08a29c6e.

**Comment by long2ice** [comment]
Fixed, just try latest dev branch

**Comment by DavideGalilei** [comment]
I just tried, and it works now 👌🏻
